src/test/java/DataDriven/AutoFillForm.java

```diff
@@ -0,0 +1,226 @@
+	  //Act
+	  for(FormModelObject expectedFormObject: expectedFormObjects) {
+		  FormModelObject actualFormObject = new FormPage(driver,url)
+				  .navigate()
+				  .fillForm(expectedFormObject)
+				  .submit()
+				  .getResults();
+		  //Assert
+		  Assert.assertTrue(actualFormObject.equals(expectedFormObject));
+	  }
+	  
+  }
+  
+  @Test
+  public void Test4() {
+	  //Arrange
+	  String url = "https://the-internet.herokuapp.com/";
+	  List<FormModelObject> expectedFormObjects = getFormObjectFromSQL();
+	  //Act
+	  for(FormModelObject expectedFormObject: expectedFormObjects) {
+		  FormModelObject actualFormObject = new FormPage(driver,url)
+				  .navigate()
+				  .fillForm(expectedFormObject)
+				  .submit()
+				  .getResults();
+		  //Assert
+		  Assert.assertTrue(actualFormObject.equals(expectedFormObject));
+	  }
+  }
+  
+  private List<FormModelObject> getFormObjectFromSQL() {
+	  List<FormModelObject> studentObjects = new ArrayList<FormModelObject>();
+	  
+	  HashMap<String,String> configs = null;
+	  try {
+		  configs = new ConfigurationReader().getPropertiesFromResourceFile("config.properties");
+	  } catch (IOException e) {
+		  //e.printStackTrace();
+		  throw new RuntimeException(e+"Config file does not exist");
+	  }
+
+	  String url = "jdbc:mysql://localhost:3306/world_x?allowPublicKeyRetrieval=true&useSSL=false";
+	  String user = "root";
+	  String password = configs.get("dbpass");
+	  
+	  String query = "SELECT first_name,last_name,isMale,phone FROM students;";
+	  try {
+		  Connection con = DriverManager.getConnection(url, user, password);
+		  Statement st = con.createStatement();
+		  ResultSet rs = st.executeQuery(query);
+		  while (rs.next()) {
+			  String first = rs.getString(1);
+			  String last = rs.getString(2);
+			  String email = "nomail@no.com";
+			  String gender = rs.getBoolean(3) ? "male":"female";
+			  String number = rs.getString(4).replace("-", "");
+			  studentObjects.add(new FormModelObject(first, last, email,  gender, number));
+		  }
+
+	  } catch (SQLException ex) {
+		  System.out.println(ex);
+	  } 
+
+	  return studentObjects;
+}
+
+private List<FormModelObject> getFormObjectFromXML() {
+	List<FormModelObject> studentObjects = new ArrayList<FormModelObject>();
+	
+	
+	try {
+		URL resource = getClass().getClassLoader().getResource("Students.xml");
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+		Document doc = dBuilder.parse(resource.getFile());
+		doc.getDocumentElement().normalize();
+		NodeList nList = doc.getElementsByTagName("person");
+		for(int i=0; i<nList.getLength();i++) {
+			Node n = nList.item(i);
+			if(n.getNodeType() == Node.ELEMENT_NODE) {
+				Element e = (Element) n;
+				String first = e.getElementsByTagName("f_name").item(0).getTextContent();
+				String last = e.getElementsByTagName("l_name").item(0).getTextContent();
+				String email = "noemail@none.org";
```

Comment:
getFormObjectFromXml( ) should attempt to find all fields that exist in the form regardless if they exist in the xml file you're reading from. Essentially this is a mapping from a Students.xml file to your Student object.